catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -1142,32 +1146,41 @@ private ElementSetNameType createElementSetName(ElementSetType type) {
         } else {
             queryType.setElementSetName(createElementSetName(ElementSetType.FULL));
         }
-        SortByType sortBy = createSortBy(query);
+        SortByType sortBy = createSortBy(queryRequest);
         if (sortBy != null) {
             queryType.setSortBy(sortBy);
         }
-        QueryConstraintType constraint = createQueryConstraint(query);
+        QueryConstraintType constraint = createQueryConstraint(queryRequest);
         if (null != constraint) {
             queryType.setConstraint(constraint);
         }
         ObjectFactory objectFactory = new ObjectFactory();
         return objectFactory.createQuery(queryType);
     }
 
-    private SortByType createSortBy(Query query) {
-
-        SortByType sortBy = null;
+    private SortByType createSortBy(QueryRequest queryRequest) {
 
-        if (query.getSortBy() != null) {
-            sortBy = new SortByType();
-            SortPropertyType sortProperty = new SortPropertyType();
-            PropertyNameType propertyName = new PropertyNameType();
+        Query query = queryRequest.getQuery();
+        SortByType cswSortBy = null;
+
+        if (query.getSortBy() != null && query.getSortBy()
+                .getPropertyName() != null) {
+            List<SortBy> sortBys = new ArrayList<>();
+            sortBys.add(query.getSortBy());
+            Serializable extSortBySer = queryRequest.getPropertyValue(EXT_SORT_BY);
+            if (extSortBySer instanceof SortBy[]) {
+                SortBy[] extSortBys = (SortBy[]) extSortBySer;
+                if (extSortBys.length > 0) {
+                    sortBys.addAll(Arrays.asList(extSortBys));
+                }
+            }
 
-            String propName;
-            if (query.getSortBy()
-                    .getPropertyName() != null) {
-                propName = query.getSortBy()
-                        .getPropertyName()
+            cswSortBy = new SortByType();
+            for (SortBy sortBy : sortBys) {
+                SortPropertyType sortProperty = new SortPropertyType();
+                PropertyNameType propertyName = new PropertyNameType();
+                String propName;
+                propName = sortBy.getPropertyName()
```

Comment:
I think this can be condensed to one line : `String propName  = sortBy.getPropertyName().getPropertyName()`.  Same concern about NPE on the second `getPropertyName()`